### PR TITLE
Add attachment permissions

### PR DESCRIPTION
**Pull request by maximilianhuber**
To download an attachment one has to say **in which context (release, project, ...) one wants to download the attachment**. Then the application checks if
- the attachment is connected to the context object and
- the user has read permissions to the context object.

The context consists of a type (i.e. a String) and an ID.

--

The word "context" is not perfect, if someone has a better idea please add a comment.

### CODE REVIEW

File: backend/utils/src/test/java/org/eclipse/sw360/attachments/db/RemoteAttachmentDownloaderTest.java [hunk 1]
```diff
@@ -83,7 +85,7 @@ public void testIntegration() throws Exception {
 
         assertThat(retrieveRemoteAttachments(DatabaseSettings.getConfiguredHttpClient(), dbName, downloadTimeout), is(1));
 
-        assertThat(attachmentConnector.getAttachmentStream(attachmentContent), hasLength(greaterThan(0l)));
+        assertThat(attachmentConnector.getAttachmentStream(attachmentContent, new User(), new Project()), hasLength(greaterThan(0l)));
```
Comment (maximilianhuber) [inline backend/utils/src/test/java/org/eclipse/sw360/attachments/db/RemoteAttachmentDownloaderTest.java]: Interesting question, I do not know.

File: libraries/lib-datahandler/src/test/java/org/eclipse/sw360/datahandler/couchdb/AttachmentStreamConnectorTest.java [hunk 2]
```diff
@@ -23,6 +26,7 @@
 import java.util.concurrent.TimeUnit;
 
 import static org.eclipse.sw360.datahandler.common.Duration.durationOf;
+import static org.hamcrest.CoreMatchers.any;
```
Comment (alex-evo) [inline libraries/lib-datahandler/src/test/java/org/eclipse/sw360/datahandler/couchdb/AttachmentStreamConnectorTest.java]: This import is unused and the references to `Mockito.any` in lines 70 and 73 are unnecessarily qualified.

**Comment by mcjaeger** [comment]
@maximilianhuber could you pls. help me more with a test scenario of I manage to have one context and then another? I remember faintly that you explained this already, but frankly I am not getting this ready now.

**Comment by maximilianhuber** [comment]
How to test:
1. create a private project as admin
2. add an attachment to the created project
3. copy the download link of the attachment
4. log out and log in as user
5. try to open copied link

**Comment by maximilianhuber** [comment]
@mcjaeger this PR is now rebased and the merge conflicts are resolved.

**Comment by mcjaeger** [comment]
download broken for all downloads, some rebase error presumeably

**Pull request merged**